File: src/main/java/com/ject/studytrip/stamp/infra/jpa/StampRepositoryAdapter.java
package com.ject.studytrip.stamp.infra.jpa;

import com.ject.studytrip.stamp.domain.model.Stamp;
import com.ject.studytrip.stamp.domain.repository.StampRepository;
import java.util.List;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class StampRepositoryAdapter implements StampRepository {
    private final StampJpaRepository stampJpaRepository;

    @Override
    public Stamp save(Stamp stamp) {
        return stampJpaRepository.save(stamp);
    }

    @Override
    public List<Stamp> saveAll(List<Stamp> stamps) {
        return stampJpaRepository.saveAll(stamps);
    }

    @Override
    public Optional<Stamp> findById(Long stampId) {
        return stampJpaRepository.findById(stampId);
    }

    @Override
    public List<Stamp> findAllByIdIn(List<Long> ids) {
        return stampJpaRepository.findAllByIdIn(ids);
    }

    @Override
    public List<Stamp> findAllByTripIdAndDeletedAtIsNull(Long tripId) {
        return stampJpaRepository.findAllByTripIdAndDeletedAtIsNull(tripId);
    }

    @Override
    public List<Stamp> findAllByTripIdOrderByDeadlineAsc(Long tripId) {
        return stampJpaRepository.findAllByTripIdOrderByDeadlineAsc(tripId);
    }
}

File: src/main/kotlin/com/ject/studytrip/stamp/infra/jpa/StampRepositoryAdapter.kt
package com.ject.studytrip.stamp.infra.jpa

import com.ject.studytrip.stamp.domain.model.Stamp
import com.ject.studytrip.stamp.domain.repository.StampRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
class StampRepositoryAdapter(
    private val stampJpaRepository: StampJpaRepository,
) : StampRepository {
    override fun save(stamp: Stamp): Stamp = stampJpaRepository.save(stamp)

    override fun saveAll(stamps: List<Stamp>): List<Stamp> = stampJpaRepository.saveAll(stamps)

    override fun findById(stampId: Long): Optional<Stamp> = stampJpaRepository.findById(stampId)

    override fun findAllByIdIn(stampIds: List<Long>): List<Stamp> = stampJpaRepository.findAllByIdIn(stampIds)

    override fun findAllByTripIdAndDeletedAtIsNull(tripId: Long): List<Stamp> = stampJpaRepository.findAllByTripIdAndDeletedAtIsNull(tripId)

    override fun findAllByTripIdOrderByCreatedAtAsc(tripId: Long): List<Stamp> =
        stampJpaRepository.findAllByTripIdOrderByCreatedAtAsc(tripId)
}
